doc/developer/design/README.md

```diff
@@ -34,21 +34,24 @@ S3 sources: https://github.com/MaterializeInc/materialize/issues/4914
 
 ## When should you make a design document?
 
-There is no hard and fast rule on when to write a design doc. But a few useful
-heuristics are:
+Some specific times when you should write a design document:
 1. If the change is large/cross-cutting, eg. will be spread over multiple PRs
-2. If the change is long and will proceed in phases that need clearly delimited scope
+2. If the change will take more than a week to implement or will proceed in phases that need clearly delimited scope
 3. If there are multiple alternative implementations and no clear best option
 4. If it's going to involve multiple people coordinating changes
+5. If it changes a customer-facing/public API, or a major private API (for example, the API for implementing new sinks)
 
 Many smaller changes do still benefit from a quick design doc to clarify
-thinking.
+thinking. Err on the side of writing a design document.
 
 ## How should you make a design document?
 
 1. Copy the template to a new date-prefixed file in `doc/developer/design` and fill it in.
 2. Submit a pull request - this makes it easy for others to add written comments.
-3. Schedule a meeting, typically in the 30-60min range. Provide time to read
-   the doc if people haven't had the opportunity, then go through it section by
-   section. If you find you need more than 60min, consider having focused meetings on
-   subsections of the design or break the design into subcomponents with separate documents.
+3. Announce that the design doc is ready for review in #eng-announce.
+
+## How should you finalize and merge a design document?
+
+1. Announce the intent to close commenting on the design document in #eng-announce.
+2. Allow two business days for any final comments.
+3. If no comments have raised new issues or if no one has asked for additional time to review, merge the design document.
```

Comment:
So one weird edge case here is that according to letter of these rules, it'd be ok to merge a design doc with no "approval" stamps. I not particularly worried about the loophole—it's not like people are going to sneak things in!—but I think it would be good to try to write down whose approval you should seek.

I'm not quite sure what the right standard is. It's something sorta like "you want explicit approval from at least one of the owners of the area that your design touches", where the owner is often the TL—but for some projects, might be much more granular than the TL. 

Here's a strawman proposal for discussion:

> For most design docs, there are key engineering or product stakeholders for the project. You should make sure to collect explicit approvals from these stakeholders before merging. If it is not clear who the stakeholders are, ask your tech lead or manager, or ask in #engineering.

Reply:
Good point! Better to be explicit instead of implicit about that. I rearranged some things and pushed a new revision, let me know if you think that covers it sufficiently 🙂